plan/column_pruning.go

```diff
@@ -157,6 +157,23 @@ func (p *Union) PruneColumns(parentUsedCols []*expression.Column) {
 // PruneColumns implements LogicalPlan interface.
 func (p *DataSource) PruneColumns(parentUsedCols []*expression.Column) {
 	used := getUsedList(parentUsedCols, p.schema)
+
+	// Resolve all generated columns.
+	var name2Offset = make(map[string]int, len(p.schema.Columns))
```

Comment:
1) In other logical plans, we don't need to resolve generated columns, so I prefer keep this logic here;
2) yes, we only needs resolve virtual generated columns.